apps/server/src/utils/fileManagement.ts

```diff
@@ -56,7 +56,7 @@ export function generateUniqueFileName(directory: string, filename: string): str
   const extension = extname(filename);
   const baseName = basename(filename, extension);
 
-  let counter = 0;
+  let counter = Number(baseName.match(/\((\d)\)$/)?.at(1) ?? 0);
```

Comment:
suggestion: we could use the `getProjectNumber` function here to minimise the amount of times we need to maintain this regex and to simplify reading the code

Reply:
ah yes, I just forgot to use it after extracting it